File: Chanseok/프로그래머스/Lv1/17681. ［1차］ 비밀지도/비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        int[] res = new int[n];
        for(int i = 0; i < n; ++i)
            res[i] = arr1[i] | arr2[i];

        String str;
        for(int i = 0; i < n; ++i) {
            str = "";
            for(int j = n-1; j >= 0; --j) {
                if((res[i] &(1 << j)) != 0) str+="#";
                else str+=" ";
            }
            answer[i] = str;
        }

        return answer;
    }
}

File: LeeGunHee/프로그래머스/비밀지도/비밀지도.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;
 
int n1[16]={0, };
int n2[16]={0, };
 
 
vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;

    for(int i = 0; i < n; i++){ 1        
        int idx = n - 1;
        while(arr1[i] != 0) {
            n1[idx--] = (arr1[i] % 2);
            arr1[i] /= 2;
        }
        idx = n-1;
        while(arr2[i] != 0){
            n2[idx--] = (arr2[i] % 2);
            arr2[i] /= 2;
        }
        
        string str;

        for(int i = 0; i < n; i++) {
            if(n1[i] || n2[i]) str += "#";
            else str += " ";
        }
        for(int i = 0; i < n; i++){
            n1[i] = 0;
            n2[i] = 0;
        }
        answer.push_back(str);
    }
    
    return answer;
}